src/main/java/org/jenkinsci/plugins/gogs/GogsWebHook.java

```diff
@@ -153,6 +153,23 @@ public void doIndex(StaplerRequest req, StaplerResponse rsp)  throws IOException
 				if (property != null) { /* only if Gogs secret is defined on the job */
 					jSecret = property.getGogsSecret(); /* Secret provided by Jenkins */
 				}
+            } 
+            else {
+                String ref = (String)jsonObject.get("ref"); 
+                String[] components = ref.split("/"); 
```

Comment:
It's pretty common for branches to be prefixed with `feature/` or `bugfix/`. A branch for example named `feature/add-something` would be detected as `add-something` with this code.

Looking at my Gogs webhook logs I see the ref is always in the format `refs/heads/feature/add-something`, so perhaps we could simply use `String branchName = ref.replace("refs/heads/", "");`. What do you think?